terraform/s3/main.tf

```
@@ -0,0 +1,134 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+}
+
+# S3 Bucket
+# Creates the main S3 bucket resource
+resource "aws_s3_bucket" "main" {
```

```suggestion
resource "aws_s3_bucket" "this" {
```

`this` is the standard resource name used in reusable Terraform modules (e.g., [`terraform-aws-modules`](https://github.com/terraform-aws-modules)) and helps keep naming consistent.